src/settings/OrderNumber.js

```diff
@@ -22,44 +14,61 @@ class OrderNumber extends Component {
 
   constructor(props) {
     super(props);
+
     this.configManager = props.stripes.connect(ConfigManager);
   }
 
   getInitialValues = (settings) => {
-    const value = settings.length && settings[0].value === 'true';
+    const value = settings.length === 0 ? '' : settings[0].value;
+    const defaultConfig = {
+      orderNumber: false,
+      selectedPrefixes: [],
+      prefixesList: [],
+      selectedSuffixes: [],
+      suffixesList: [],
+    };
+    let config;
+
+    try {
+      config = Object.assign({}, defaultConfig, JSON.parse(value));
```

Comment:
Actually it's better to leave in try-catch block only code with parsing error expected:
```
orderNumberSetting = JSON.parse(orderNumberSetting)
```
in catch you can assign `{}` to `orderNumberSetting`.
and after just make `Object.assign(config, orderNumberSetting);`